app/containers/EmojiPicker/EmojiCategory.js

```diff
@@ -42,19 +44,70 @@ export default class EmojiCategory extends React.Component {
 
 		this.size = Math.min(width || widthWidth, windowHeight) / (emojisPerRow || EMOJIS_PER_ROW);
 		this.emojis = props.emojis;
+		this.state = {
+			isModalVisible: false,
+			modalX: 0,
+			modalY: 0,
+			modalData: []
+		};
 	}
 
-	shouldComponentUpdate() {
+	shouldComponentUpdate(nextProps, nextState) {
+		const { isModalVisible } = this.state;
+		if (isModalVisible !== nextState.isModalVisible) {
+			return true;
+		}
 		return false;
 	}
 
+	componentWillUnmount() {
+		this.setState({ isModalVisible: false });
+	}
+
+	getModalStyle() {
+		const { modalX, modalY } = this.state;
+		const { width } = Dimensions.get('window');
+		const widthOfModal = this.size * 5.2;
+		let xCenter = modalX - (widthOfModal) / 2;
+		if (xCenter < 0) {
+			xCenter = 0;
+		} else if (xCenter + (widthOfModal) / 2 > widthOfModal) {
+			xCenter = width - widthOfModal - 20;
+		}
+
+		const yCenter = modalY - 2.5 * this.size;
+		return {
+			position: 'absolute',
+			alignSelf: 'baseline',
+			width: widthOfModal,
+			left: xCenter,
+			top: yCenter,
+			flexDirection: 'row',
+			backgroundColor: '#fefefe',
+			borderRadius: 10,
+			padding: 5,
+			borderColor: '#d6d7da',
+			borderWidth: 0.5
+		};
+	}
+
+	showModal = (event, modalData) => {
+		this.setState({
+			isModalVisible: true,
+			modalX: event.nativeEvent.pageX,
+			modalY: event.nativeEvent.pageY,
```

Comment:
@diegolmello issue arises because of this as documentation says([https://facebook.github.io/react-native/docs/gesture-responder-system
](https://facebook.github.io/react-native/docs/gesture-responder-system))this should potentially give The Y position of the touch, relative to the root element it as working as expected in android but in case of IOS it is taking message-box top as zero and giving Y relative to that, now I tried to find out the height of the message box so I can find th e offset by subtracting it from total height but doing that seems impossible from emojiCategory as it is returning total height of list instead of its displayed height

Reply:
@pranavpandey1998official Forget about this PR for now and work on other issues then. I'll try to find an alternative soon.